addons/mail/static/tests/new/views/chatter_topbar_tests.js

```diff
@@ -180,3 +181,85 @@ QUnit.test("log note/send message switching", async function (assert) {
     );
 });
+
+QUnit.test("attachment counter without attachments", async function (assert) {
+    const pyEnv = await startServer();
+    const resPartnerId = pyEnv["res.partner"].create({});
+    const { openView } = await start();
+    await openView({
+        res_id: resPartnerId,
+        res_model: "res.partner",
+        views: [[false, "form"]],
+    });
+    assert.containsOnce(target, ".o-mail-chatter-topbar-add-attachments");
+    assert.containsOnce(target, ".o-mail-chatter-topbar-add-attachments:contains(0)");
+});
+
+QUnit.test("attachment counter with attachments", async function (assert) {
+    const pyEnv = await startServer();
+    const resPartnerId = pyEnv["res.partner"].create({});
+    pyEnv["ir.attachment"].create([
+        {
+            mimetype: "text/plain",
+            name: "Blah.txt",
+            res_id: resPartnerId,
+            res_model: "res.partner",
+        },
+        {
+            mimetype: "text/plain",
+            name: "Blu.txt",
+            res_id: resPartnerId,
+            res_model: "res.partner",
+        },
+    ]);
+    const { openView } = await start();
+    await openView({
+        res_id: resPartnerId,
+        res_model: "res.partner",
+        views: [[false, "form"]],
+    });
+    assert.containsOnce(target, ".o-mail-chatter-topbar-add-attachments:contains(2)");
+});
+
+QUnit.test("attachment counter while loading attachments", async function (assert) {
+    const pyEnv = await startServer();
+    const resPartnerId1 = pyEnv["res.partner"].create({});
+    const { openView } = await start({
+        async mockRPC(route) {
+            if (route.includes("/mail/thread/data")) {
+                await makeDeferred(); // simulate long loading
+            }
+        },
+    });
+    await openView({
+        res_id: resPartnerId1,
+        res_model: "res.partner",
+        views: [[false, "form"]],
+    });
+    assert.containsOnce(target, ".o-mail-chatter-topbar-add-attachments .fa-spin");
+    assert.containsNone(target, ".o-mail-chatter-topbar-add-attachments:contains(0)");
+});
+
+QUnit.test("attachment counter transition when attachments become loaded", async function (assert) {
+    const pyEnv = await startServer();
+    const resPartnerId1 = pyEnv["res.partner"].create({});
+    const attachmentPromise = makeDeferred();
+    const { openView } = await start({
+        async mockRPC(route) {
+            if (route.includes("/mail/thread/data")) {
+                await attachmentPromise;
+            }
+        },
+    });
+    await openView({
+        res_id: resPartnerId1,
+        res_model: "res.partner",
+        views: [[false, "form"]],
+    });
+    assert.containsOnce(target, ".o-mail-chatter-topbar-add-attachments .fa-spin");
+    assert.containsNone(target, ".o-mail-chatter-topbar-add-attachments:contains(0)");
+
+    await afterNextRender(() => attachmentPromise.resolve());
+    assert.containsNone(target, ".o-mail-chatter-topbar-add-attachments .fa-spin");
```

Comment:
shouldn't icon change after attachment become loaded?

Reply:
`fa-paperclip` stays regardless of loading or not. It's either spinner or textual counter. I use `:contains()` for asserting textual counter